### PR TITLE
Egt sensors to SensorType framework

### PR DESCRIPTION
**Pull request by dron0gus**

### CODE REVIEW

**Comment by dron0gus** [comment]
@zed65 can you test that nothing was broken around EGT/MAX31855?

File: firmware/hw_layer/sensors/max31855.cpp [hunk 1]
```diff
@@ -47,7 +44,7 @@ class Max31855Read final : public ThreadController<UTILITY_THREAD_STACK_SIZE> {
 		MC_OPEN = 2,
 		MC_SHORT_GND = 3,
 		MC_SHORT_VCC = 4,
-	} max_32855_code;
+	} max_31855_code;
```
Comment (rusefillc) [inline firmware/hw_layer/sensors/max31855.cpp:47]: Ой :)

**Comment by zed65** [comment]
Do you want me to test current master or should I wait on this PR?

**Comment by zed65** [comment]
Current master EGT works.

**Comment by rusefillc** [comment]
Just merged I guess we would have to test fresh master :)

**Comment by zed65** [comment]
Nope, EGT sits at 23C no matter what.

**Pull request closed without merging**